backend/src/controllers/review.controller.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  IReview,
+  ISentimentBarChartGroup,
+  IWordFrequency,
+  IPieChartData,
+  Sentiment,
+  IBusinessAnalysis,
+} from 'shared-types';
+import ReviewModel from '../models/review.model';
+import { BaseController } from './base.controller';
+import { AuthRequest } from 'common/auth.middleware';
+import httpStatus from 'http-status';
+import { Response } from 'express';
+
+class ReviewController extends BaseController<IReview> {
+  constructor() {
+    super(ReviewModel);
+  }
+
+  async getAnalysis(req: AuthRequest, res: Response) {
+    const { businessId } = req.user!;
+    const today = new Date();
+    today.setHours(0, 0, 0, 0);
+    const sevenDaysAgo = new Date(today.getTime() - 7 * 24 * 60 * 60 * 1000);
```

Comment:
Seperate to another file

```suggestion
const daysAgo = (days) => {
    const today = new Date();
    today.setHours(0, 0, 0, 0);
    return new Date(today.getTime() - days * 24 * 60 * 60 * 1000);
}
```